pkg/identity/service.go

```diff
@@ -104,16 +106,19 @@ func (svc *Service) Run(issuerEvent <-chan struct{}, issuerError <-chan error) {
 }
 
 func (svc *Service) loadCredentials() (tls.Issuer, error) {
```

Comment:
can you explain the change in this function?  I'm not clear on what the difference is between `ReadPEMCreds` followed by `Verify` vs `LoadIssuerCrtAndKeyFromFiles` followed by `VerifyAndBuildCred`.

Reply:
Yes, so I wanted to consolidate the more "bespoke" verification logic we have into `VerifyAndBuildCred`. Namely this one also checks for algorith requirements + whether the issuer is a CA (will check for that...) + provides more explicit description around potential problems with time validity. This is the reason.

Reply:
Can any of the "bespoke" methods be removed?  It's not clear to me why we have several different methods which do basically the same thing.

Reply:
Certainly agree. Just added an issue for that that I will address right after these three PRs are merged as this would touch all of them most likely. Is that fine? 

https://github.com/linkerd/linkerd2/issues/3828